scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/kafka/KafkaAdmin.kt

```diff
@@ -46,12 +68,27 @@ class KafkaAdmin(
                     )
                 }
                 .run {
-                    adminClient.createTopics(this)
+                    adminClient.createTopics(
+                        this,
+                        CreateTopicsOptions().timeoutMs(topicWaitRetryParams.createTimeoutMillis)
+                    )
                 }
                 .values()
                 .also { topicCreations ->
-                    topicCreations.entries.forEach { creationResult ->
-                        awaitKafkaResult(creationResult)
+                    logger.info("Waiting for kafka topic creation")
+                    // We repeatedly attempt to describe all topics as a way of blocking until they exist at least on
+                    // one broker. This is because the call to createTopics above returns before topics can actually
+                    // be subscribed to.
+                    retry(
+                        missingTopicRetryPolicy + limitAttempts(topicWaitRetryParams.describeRetries) + constantDelay(
+                            topicWaitRetryParams.describeRetryDelayMillis
+                        )
+                    ) {
+                        logger.debug("Still waiting for all topics to be created...")
+                        // the KafkaFuture retrieved via .allTopicNames() only succeeds if all the topic
+                        // descriptions succeed, so there is no need to check topic descriptions individually
+                        adminClient.describeTopics(topicCreations.keys).allTopicNames()
```

Comment:
just to confirm that `allTopicNames()` is only for the current topology?  (lack of my understanding of the syntax possibly of the flow).

Reply:
Indeed, only for the current topology. `topicCreations` is essentially a map where the topic names for the current topology (from `steps`) are keys. `.allTopicNames()` tries to get all the topic names for the list of topics passed to `describeTopics`